addons/stock/models/stock_move_line.py

```diff
@@ -586,7 +587,9 @@ def _free_reservation(self, product_id, location_id, quantity, lot_id=None, pack
                     candidate.product_uom_qty = self.product_id.uom_id._compute_quantity(quantity_split, candidate.product_uom_id, rounding_method='HALF-UP')
                     move_to_recompute_state |= candidate.move_id
                     break
-            move_to_recompute_state._recompute_state()
+            to_unlink_candidates = self.env['stock.move.line'].browse(to_unlink_candidate_ids)
+            (move_to_recompute_state - to_unlink_candidates.move_id)._recompute_state()
```

Comment:
Instead of remove move of `to_unlink_candidates` of move_to_recompute_state, why not just moving `move_to_recompute_state |= candidate.move_id` (the first one) two lines below.

Reply:
Thats..actually what I did in PR odoo/odoo#73082 😟 I forgot that I didn't forget to implement Whenrow's suggestion 😟😟